doc/book/quick-start-skeleton.md

```diff
@@ -0,0 +1,259 @@
+`src/Action/HelloAction.php`:
+
+```php
+<?php
+namespace App\Action;
+
+class HelloAction
+{
+    public function __invoke($request, $response, callable $next)
+    {
+        $query  = $request->getQueryParams();
+        $target = isset($query['target']) ? $query['target'] : 'World';
+        $target = htmlspecialchars($target, ENT_HTML5, UTF-8);
+
+        $response->getBody()->write(sprintf(
+            '<h1>Hello, %s!</h1>',
+            $target
+        ));
+        return $response->withHeader('Content-Type', 'text/html');
+    }
+}
+```
+
+The above looks for a query string parameter "target", and uses its value to
+provide a message, which is then returned in an HTML response.
+
+Now we need to inform the application of this middleware, and indicate what
+path will invoke it. Open the file `config/autoload/routes.global.php`. Inside
+that file, you should have a structure similar to the following:
+
+```php
+return [
+    'dependencies' => [
+        /* ... */
+    ],
+    'routes' => [
+        /* ... */
+    ],
+];
+```
+
+We're going to add an entry under `routes`:
+
+```php
+return [
+    /* ... */
+    'routes' => [
+        /* ... */
+        [
+            'name' => 'hello',
+            'path' => '/hello',
+            'middleware' => App\Action\HelloAction::class,
+            'allowed_methods' => ['GET'],
+        ],
+    ],
+];
+```
+
+Once you've added the above entry, give it a try by going to each of the
+following URIs:
+
+- http://localhost:8080/hello
+- http://localhost:8080/hello?target=ME
+
+You should see the message change as you go between the two URIs!
+
+### Using templates
+
+You likely don't want to hardcode HTML into your middleware; so, let's use
+templates. This particular exercise assumes you chose to use the
+[Plates](http://platesphp.com) integration.
+
+Templates are installed under the `templates/` subdirectory. By default, we also
+register the template namespace `app` to correspond with the `templates/app`
+subdirectory. Create the file `templates/app/hello-world.phtml` with the
+following contents:
+
+```php
+<?php $this->layout('layout::default', ['title' => 'Greetings']) ?>
+
+<h2>Hello, <?= $this->e($target) ?></h2>
+```
+
+Now that we have a template, we need to:
+
+- Inject a renderer into our action class.
+- Use the renderer to render the contents.
+
+Replace your `src/Action/HelloAction.php` file with the following contents:
+
+```php
+<?php
+namespace App\Action;
+
+use Zend\Diactoros\Response\HtmlResponse;
+use Zend\Expressive\Template\TemplateRendererInterface;
+
+class HelloAction
+{
+    private $renderer;
+
+    public function __construct(TemplateRendererInterface $renderer)
+    {
+        $this->renderer = $renderer;
+    }
+
+    public function __invoke($request, $response, callable $next)
+    {
+        $query  = $request->getQueryParams();
+        $target = isset($query['target']) ? $query['target'] : 'World';
+
+        return new HtmlResponse(
```

Comment:
I know creating a new `HtmlResponse` is very convenient, but I'm not sure if this is a good thing. Also, in previous example we was using the $response provided via argument.